src/coreclr/tools/Common/TypeSystem/Mutable/MutableModule.Sorting.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.TypeSystem.Ecma
+{
+    partial class MutableModule
+    {
+        // This isn't deterministic, but it is functional. At this time, since only 1 MutableModule is contained in a build, it will be deterministic as it will always return 0.
```

Comment:
I already felt that maybe this shouldn't in the common type system, but Tomas's comment brought me over the fence.

Determinism and thread safety are the core invariants in the type system.

Could all of this just go to ILCompiler.ReadyToRun? It doesn't really look reusable.